c/charsets.c

```diff
@@ -200,22 +200,24 @@ int convertCharset(char *input,
 }
 
 int getCharsetCode(char *charsetName) {
-  if (!strcmp(charsetName, "ISO-8859-1")) {
+  if ((!strcmp(strupcase(charsetName), "ISO-8859-1"))  ||
```

Comment:
What if someone passes a string literal? That would cause an S0C4 ABEND.

For example,
```c
getCharsetCode("IBM_1047");
```

I suggest making the argument `const` and using a local buffer.

As for the uppercasing, is passing mixed case names valid? Wouldn't it be better to uppercase before passing the name?

Reply:
Code was modified to use local array in comparison.

The other functions that we have modified/created allows for these encoding to be case insensitive.  The strupcase is keeping with this policy.